### ISSUE TITLE
ProjectedAdaptiveLogSoftmax.log_prob raises Exception

### ISSUE THREAD

**Issue by klicperajo** [huggingface/transformers#4461]
# 🐛 Bug

## Information

Model I am using (Bert, XLNet ...): Transformer-XL

Language I am using the model on (English, Chinese ...): WikiText-103 (English)

The problem arises when using:
* [x] the official example scripts: `run_transfo_xl.py` (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [x] an official task: WikiText-103
* [ ] my own task or dataset: (give details below)

## To reproduce

1. Fix `run_transfo_xl.py` by removing the unused `work_dir` argument (line 52) and changing `lm_labels=target` to `labels=target` (line 108).
2. Add `logits = self.crit.log_prob(pred_hid.flatten(0, 1))` right before the model output, e.g. in line 919 of `modeling_transfo_xl.py`.
3. Run `run_transfo_xl.py`.
4. Look at error message:
```
The size of tensor a (1280) must match the size of tensor b (20000) at non-singleton dimension 1
  File ".../transformers/src/transformers/modeling_transfo_xl_utilities.py", line 246, in log_prob
    logprob_i = head_logprob[:, -i] + tail_logprob_i
  File ".../transformers/src/transformers/modeling_transfo_xl.py", line 920, in forward
    logits = self.crit.log_prob(pred_hid.flatten(0, 1))
  File ".../run_transfo_xl.py", line 107, in evaluate
    ret = model(data, labels=target, mems=mems)
  File ".../run_transfo_xl.py", line 124, in main
    test_loss = evaluate(te_iter)
  File ".../run_transfo_xl.py", line 143, in <module>
    main()
```

## Expected behavior

`log_prob` should return the log probabilities instead of raising an Exception.

## Environment info

     
- `transformers` version: Master (commit 384f0eb)
- Platform: Ubuntu
- Python version: 3.7.6
- PyTorch version (GPU?): 1.4.0 (CUDA 10.1, CuDNN 7.6.3)
- Tensorflow version (GPU?): -
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No
